### PR TITLE
add support for multiple infraenv per cluster

### PR DESCRIPTION
**Pull request by rccrdpccl**
Add support for multiple infraenv.
It will be now an attribute of hosts (`.cluster.hosts[].infra_env`)
Also a summary will be generated, in order to select clusters based on infra_env properties: `cluster.hosts_summary`

This is done to enable statistical observation of multiarch heterogeneous clusters

### CODE REVIEW

File: assisted-events-scrape/tests/integration/test_integration.py [hunk 1]
```diff
@@ -68,44 +67,51 @@ def check_document_count(index, expected_count):
...
+        random_event = self._get_random_event_for_cluster("d386f7df-03ba-46bf-a49b-f6b65a0fb90d")
+
+        cluster = random_event["cluster"]
+        assert_cluster_data(cluster)
+        assert_hosts_data(cluster)
+        assert_host_summary_data(cluster)
+        assert_is_not_multiarch(cluster)
+        assert_cluster_iso_type(cluster, "full-iso")
+
+        random_event = self._get_random_event_for_cluster("fc124c08-f3a5-464f-834e-c262d9eb5a38")
```
[the remark this answers is not in the document]
Comment (mkowalski) [reply assisted-events-scrape/tests/integration/test_integration.py:80]: Please note `"cpu_architecture": "multiarch",` should be `"cpu_architecture": "multi"

[the remark this answers is not in the document]
Comment (mkowalski) [reply assisted-events-scrape/tests/integration/test_integration.py:80]: (there was one commit when it was indeed `multiarch` but we changed it quickly and the current API defines it as `multi`)

[the remark this answers is not in the document]
Comment (rccrdpccl) [reply assisted-events-scrape/tests/integration/test_integration.py:80]: Thanks! I've redefined the tests to mock the value as `multi` so we do not create unnecessary confusion.

For marking it as heterogeneous this value is not taken into account: a cluster is heterogeneous if it has hosts of different arch, independently of its own `cpu_architecture` value.

File: assisted-events-scrape/process/host.py [hunk 2]
```diff
@@ -0,0 +1,66 @@
+
+class Host:
```
Comment (adriengentil) [inline assisted-events-scrape/process/host.py:2]: nit: is the class really needed? There is only one class method and the rest are all free functions.

File: assisted-events-scrape/process/host.py [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+    to_summarize = {
```
Comment (adriengentil) [inline assisted-events-scrape/process/host.py:3]: nit: the variable should be in upper case for f it's used as a const, and prefixed with `_` as it looks private too

**Review by adriengentil — COMMENTED** [review]
Just 2 small comments, looks good overall :+1:

**Review by adriengentil — APPROVED** [review]

**Pull request merged**